Bottom: ebe2a8e04681c0de82b4c1f3fff9f316a6e9205c
Top:    68e77cf1173ac67e41f6c8c46334b19059b2e47a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-25 02:27:10 +0100

Refresh of fix-mesa-crash-on-startup

---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index ea2c783..5b52b9a 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -111,6 +111,13 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
   return FALSE;
 }
 
+
+void
+ghid_port_drawing_realize_cb (GtkWidget *widget, gpointer data)
+{
+}
+
+
 void
 ghid_pinout_preview_init (GhidPinoutPreview *preview)
 {
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 9a3fb8a..0ebb64d 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -223,6 +223,22 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
 }
 
 void
+ghid_port_drawing_realize_cb (GtkWidget *widget, gpointer data)
+{
+  GdkGLContext *glcontext = gtk_widget_get_gl_context (widget);
+  GdkGLDrawable *gldrawable = gtk_widget_get_gl_drawable (widget);
+
+  /*** OpenGL BEGIN ***/
+  if (!gdk_gl_drawable_gl_begin (gldrawable, glcontext))
+    return;
+
+  gdk_gl_drawable_gl_end (gldrawable);
+  /*** OpenGL END ***/
+
+  return;
+}
+
+void
 ghid_pinout_preview_init (GhidPinoutPreview *preview)
 {
   struct render_priv *priv = gport->render_priv;
diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index c4b2c97..7c5823f 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -2143,22 +2143,6 @@ destroy_chart_cb (GtkWidget * widget, GHidPort * port)
   gtk_main_quit ();
 }
 
-static void
-ghid_port_drawing_realize_cb (GtkWidget *widget, gpointer data)
-{
-  GdkGLContext *glcontext = gtk_widget_get_gl_context (widget);
-  GdkGLDrawable *gldrawable = gtk_widget_get_gl_drawable (widget);
-
-  /*** OpenGL BEGIN ***/
-  if (!gdk_gl_drawable_gl_begin (gldrawable, glcontext))
-    return;
-
-  gdk_gl_drawable_gl_end (gldrawable);
-  /*** OpenGL END ***/
-
-  return;
-}
-
 /* 
  * Create the top_window contents.  The config settings should be loaded
  * before this is called.
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 1550003..85e5702 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -489,6 +489,7 @@ void ghid_end_drawing (GHidPort *port);
 void ghid_drawing_area_configure_hook (GHidPort *port);
 void ghid_screen_update (void);
 gboolean ghid_drawing_area_expose_cb (GtkWidget *, GdkEventExpose *, GHidPort *);
+void ghid_port_drawing_realize_cb (GtkWidget *, gpointer);
 void ghid_pinout_preview_init (GhidPinoutPreview *preview);
 gboolean ghid_pinout_preview_expose (GtkWidget * widget, GdkEventExpose * ev);
 GdkPixmap *ghid_render_pixmap (int cx, int cy, double zoom, int width, int height, int depth);
